validator/src/commands/repair_shred_from_peer/mod.rs

```diff
@@ -37,19 +56,96 @@ pub fn command(_default_args: &DefaultArgs) -> App<'_, '_> {
 }
 
 pub fn execute(matches: &ArgMatches, ledger_path: &Path) {
-    let pubkey = value_t!(matches, "pubkey", Pubkey).ok();
-    let slot = value_t_or_exit!(matches, "slot", u64);
-    let shred_index = value_t_or_exit!(matches, "shred", u64);
+    let repair_shred_from_peer_args = RepairShredFromPeerArgs::from_clap_arg_match(matches);
+    if repair_shred_from_peer_args.slot.is_none() {
+        eprintln!("slot is required");
```

Comment:
if these are required, why are they specified as optional in struct itself? seems like we should make invalid argument combinations unrepresentable, this way all the validation can be delegated to clap once we upgrade to v4.